### ISSUE TITLE
# ci: add lightweight CLI compatibility check for @lint-md/core

### ISSUE THREAD

**Issue by luojiyin1987** [lint-md/cli#102]
## Background

`@lint-md/cli` uses `@lint-md/core` as its lint and fix engine.

Normal CLI CI already tests the version of `@lint-md/core` selected by `package-lock.json`. However, it does not detect whether an upcoming change on the core default branch will break the current CLI before core is published.

This issue adds one lightweight compatibility check without building a full cross-version compatibility matrix.

## Goal

Add an informational GitHub Actions workflow that tests the CLI default branch against a package built from the current default branch of `lint-md/lint-md`.

The workflow should exercise the real packed core package rather than importing core source files directly.

## Proposed workflow

Add:

```text
.github/workflows/core-compatibility.yml
```

The workflow should support:

```yaml
workflow_dispatch:
```

It may also run on a small schedule, such as once per week, if ongoing drift detection is useful.

The workflow is informational and should not initially be required by branch protection.

## Test flow

```text
checkout lint-md/cli
        ↓
checkout lint-md/lint-md
        ↓
install, build and pack @lint-md/core
        ↓
install the generated core tarball into CLI
        ↓
verify the installed core path and version
        ↓
run the CLI compatibility smoke tests
```

Suggested commands:

```bash
# Build core
cd lint-md
npm ci
npm run build
npm pack --pack-destination ../artifacts

# Install the packed core into CLI
cd ../cli
npm ci
npm install --no-save ../artifacts/lint-md-core-*.tgz

# Verify the intended package is installed
node -e "
const path = require.resolve('@lint-md/core/package.json');
const pkg = require(path);
console.log({ path, version: pkg.version });
"

# Run compatibility checks
npm run lint
npm test
npm run build
npm run test:package
```

Adjust the exact commands to match the scripts available in each repository.

## Required compatibility coverage

The existing CLI tests or a small dedicated smoke test should verify:

* linting a Markdown file;
* applying a fix;
* linting and fixing stdin;
* running through the Piscina worker path;
* consuming `lintResult`, `fixedResult` and `executionErrors`;
* loading the packed core package successfully.

Do not duplicate the complete CLI test suite inside the workflow. Reuse existing scripts and tests.

## Package verification

The workflow must print:

* resolved `@lint-md/core` package path;
* installed core version;
* core commit SHA;
* generated tarball filename.

The job should fail if the registry or lockfile-selected core remains installed instead of the generated tarball.

## Permissions

Use read-only permissions:

```yaml
permissions:
  contents: read
```

The workflow must not require:

* npm publishing tokens;
* repository write access;
* `pull_request_target`;
* deployment credentials.

## Non-goals

This issue does not add:

* minimum-version and latest-version matrices;
* testing every published core release;
* testing parser internals;
* a full Node.js version matrix;
* bidirectional workflows in both repositories;
* automatic package publishing;
* temporary prerelease versions;
* duplicated implementations of existing CLI tests.

Normal CLI CI remains responsible for testing the dependency selected by its lockfile.

The core repository may separately test core pull requests against the CLI default branch.

## Acceptance criteria

* [ ] `core-compatibility.yml` exists.
* [ ] The workflow supports manual execution.
* [ ] Core is built and installed from an npm tarball.
* [ ] The installed core path and version are verified.
* [ ] CLI lint, fix, stdin and worker paths are exercised.
* [ ] CLI lint, test, build and package smoke checks pass.
* [ ] The workflow uses read-only permissions.
* [ ] The workflow is informational rather than required.
* [ ] The workflow does not require secrets.

**Comment by luojiyin1987** [comment]
Closing this for now.

The core and CLI repositories are personally maintained and change relatively infrequently. A scheduled cross-repository compatibility workflow would add ongoing CI usage and maintenance overhead without enough corresponding value.

Compatibility can instead be checked manually before relevant core releases or after changes to the public result/package contract by packing the candidate core package, installing it into the CLI, and running the existing CLI test and package-smoke suites.

We can reconsider an automated workflow if core/CLI development becomes more frequent or compatibility regressions become a recurring problem.
